pkg/archive/diff_unix.go

```diff
@@ -0,0 +1,15 @@
+//go:build !windows
+// +build !windows
+
+package archive
+
+import "golang.org/x/sys/unix"
+
+// overrideUmask sets current process's file mode creation mask to newmask
+// and returns a function to restore it.
```

Comment:
Document why this function is dangerous in a multithreaded process to dissuade anyone from thinking it is possible to safely change and restore the umask in Go programs.